Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Publish-UpgradedServiceFabricApplication.ps1

```diff
@@ -373,9 +371,9 @@ function Publish-UpgradedServiceFabricApplication
 
         Write-Host (Get-VstsLocString -Key SFSDK_WaitingForUpgrade)
         $upgradeStatusFetcher = { Get-ServiceFabricApplicationUpgradeAction -ApplicationName $ApplicationName }
-        $upgradeStatusValidator = { param($upgradeStatus) return ($upgradeStatus.UpgradeState -eq "RollingBackCompleted" -or $upgradeStatus.UpgradeState -eq "RollingForwardCompleted") }
+        $upgradeRetryEvaluator = { param($upgradeStatus) return ($upgradeStatus.UpgradeState -ne "RollingBackCompleted" -and $upgradeStatus.UpgradeState -ne "RollingForwardCompleted") }
```

Comment:
It is consistent. In upgradeRetryEvaluator we are checking the result to decide whether to retry. In case of Register, we need to call Get-ServiceFabricApplicationType only in case of exception and hence that is handled in $exceptionRetryEvaluator

Reply:
I mean to point out the logic of calling Invoke-ActionWithRetries in this file.
This call should ideally be done in Get-ServiceFabricApplicationUpgradeAction to as we've done for all other cmdlets.
I've raised a quick draft PR #7593 for the same, let me know if this seems fine to you so that I can go ahead test out the change and merge